src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -1380,9 +1503,44 @@ func (s *commitLogSource) ReadIndex(
 		}
 	}
 
+	shouldReturnUnfulfilled, err := s.shouldReturnUnfulfilled(
+		encounteredCorruptData, ns, shardsTimeRanges, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	if shouldReturnUnfulfilled {
+		indexResult.SetUnfulfilled(shardsTimeRanges)
```

Comment:
We still return our data right? (I know this is likely true, just want to verify).  That way we can join whatever we had at least.

Reply:
Yeah this should have no effect on what data we return (pretty sure my prop test still verifies the data is there) so as long as the caller doesn't ignore the data it will get merged in